### DISCUSSION TITLE
Duckdb Support

### DISCUSSION THREAD

**Discussion by heejit** [sqlpage/SQLPage#226] [General]
Is there any future plan to add Duckdb support (duckdb.org)

**Comment by lovasoa** [comment]
Hello and welcome to SQLPage!

DuckDB is not supported yet, but implementing it would be possible, by adding it to https://docs.rs/sqlx-oldapi

Can you tell me more about your use case? 

Duckdb has the ability to read and write to SQLite database files; wouldn't that fit your use case?

**Reply by heejit** [reply]
Duckdb can read and write csv, parquet, postgresql, sqlite and lot of other features use full for data analysis/science.

**Reply by lovasoa** [reply]
What is your use case? Do you have a particular application in mind you are trying to build?

**Reply by heejit** [reply]
No

**Comment by heejit** [comment]
Some time I work with CSV files.

**Reply by lovasoa** [reply]
Depending on your use case, I can recommend:
 - using a SQLite csv extension, such as: https://github.com/asg017/sqlite-xsv
 - using SQLPage itself to import a csv into a database : see "Bulk data insertion" in https://sql.ophir.dev/documentation.sql?component=form#component

**Comment by swasheck** [comment]
duckdb is an increasingly popular not-quite-big-data analytics tool that has multiple excellent built-in extensions so i can see it being an interesting use-case for sqlpage's unique (and quite excellent) approach to data presentation.

**Reply by lovasoa** [reply]
Hello! This has been discussed before, and DuckDB (and other analytics databases) support is definitely on my to-do list! 

Would you be interested in joining the implementation effort?
